lib/errors.d.ts

```diff
@@ -69,6 +69,7 @@ export declare class SerializationError extends OpenSearchClientError {
   message: string;
   data: any;
   constructor(message: string, data: any);
+  constructor(message: string);
```

Comment:
We really need testing for typescript definitions, and/or rewrite this whole thing in TypeScript. I'm just thinking out loud, maybe you have ideas @timursaurus of what we could do to catch more of these problems?

Reply:
I had originally planned to convert files to TypeScript gradually, folder by folder, starting with `/lib`. However, I know some hacks for identifying and addressing these issues without necessarily switching to TypeScript outright, while still using JavaScript, like the `@ts-check` flag, etc. 
`Serializer.js` just happened to be place I started from. Couldn't resist of creating PR asap :D 

I can link https://github.com/opensearch-project/opensearch-js/issues/269 in next PRs.